examples/hal/quad/main.rs

```diff
@@ -375,11 +375,10 @@ fn main() {
     let upload_size = (height * row_pitch) as u64;
     println!("upload row pitch {}, total size {}", row_pitch, upload_size);
 
-    let image_upload_memory = device.allocate_memory(upload_type, upload_size).unwrap();
-    let image_upload_buffer = {
-        let buffer = device.create_buffer(upload_size, image_stride as u64, buffer::Usage::TRANSFER_SRC).unwrap();
-        device.bind_buffer_memory(&image_upload_memory, 0, buffer).unwrap()
-    };
+    let image_buffer_unbound = device.create_buffer(upload_size, image_stride as u64, buffer::Usage::TRANSFER_SRC).unwrap();
+    let image_mem_reqs = device.get_buffer_requirements(&image_buffer_unbound);
+    let image_upload_memory = device.allocate_memory(upload_type, image_mem_reqs.size).unwrap();
```

Comment:
Just noticing that `upload_type` might not be supported by `image_buffer_unbound` as they were created with different usage flags, even though it probably works on 99% of the hardware out there.

Reply:
Indeed... but this is only slightly touching on this PR